backend/lib/kubernetes-client/resources/Resource.js

```diff
@@ -0,0 +1,167 @@
+      } else {
+        pathname += 'namespaces/' + encodeURIComponent(namespace) + '/'
+      }
+    }
+    pathname += this.constructor.names.plural
+    if (name) {
+      pathname += '/'
+      pathname += Array.isArray(name) ? name.map(encodeURIComponent).join('/') : encodeURIComponent(name)
+    }
+    return pathname
+  }
+
+  [http.patch] ({ type = 'merge', ...options } = {}) {
+    return super[http.patch](setPatchType(options, type))
+  }
+
+  [ws.create] ({ allNamespaces, namespace, name, query, headers, origin, ...options } = {}) {
+    const {
+      prefixUrl,
+      ca,
+      key,
+      cert,
+      servername,
+      rejectUnauthorized,
+      headers: defaultHeaders
+    } = this[http.client].defaults.options
+    const url = new URL(prefixUrl)
+    origin = origin || url.origin
+    url.protocol = url.protocol.replace(/^http/, 'ws')
+    url.pathname = join(url.pathname, this[http.pathname]({ allNamespaces, namespace }))
+    const searchParams = new URLSearchParams(query)
+    searchParams.set('watch', true)
+    if (name) {
+      let fieldSelector = `metadata.name=${name}`
+      if (searchParams.has('fieldSelector')) {
+        fieldSelector += ',' + searchParams.get('fieldSelector')
+      }
+      searchParams.set('fieldSelector', fieldSelector)
+    }
+    url.search = searchParams.toString()
+    headers = { ...defaultHeaders, ...headers }
+    const websocketOptions = {
+      origin,
+      servername,
+      headers,
+      key,
+      cert,
+      ca,
+      rejectUnauthorized,
+      ...options
+    }
+    return new WebSocket(url, websocketOptions)
+  }
+
+  [ws.connect] ({ timeout = 5000, retries = 2, ...options } = {}) {
+    const connect = () => pEvent(this[ws.create](options), 'open', { timeout })
+    return pRetry(connect, { retries })
+  }
+
+  [ws.watch] (options = {}) {
+    const {
+      initialDelay,
+      maxDelay,
+      strategy,
+      failAfter,
+      randomisationFactor,
+      immediate,
+      ...connectOptions
+    } = options
+
+    const reconnectDefaults = {
+      initialDelay: 5e2,
+      maxDelay: 15e3,
+      strategy: 'fibonacci',
+      failAfter: Infinity,
+      randomisationFactor: 0,
+      immediate: false
+    }
+
+    const reconnectOptions = {
+      ...reconnectDefaults,
+      initialDelay,
+      maxDelay,
+      strategy,
+      failAfter,
+      randomisationFactor,
+      immediate
+    }
+
+    const resourceName = this.constructor.names.plural
+
+    const createConnection = options => {
+      const connection = new EventEmitter()
+      connection.resourceName = resourceName
+      wrapWebSocket(connection, this[ws.create](options))
+      return connection
+    }
+
+    const onConnect = connection => {
+      connection.on('event', event => reconnector.emit('event', event))
+    }
+
+    const reconnector = inject(createConnection)(reconnectOptions, onConnect)
```

Comment:
create reconnectorClass
- include waitFor function
- include wrapWebSocket

Reply:
Created `WatchBuilder` class which contains `waitFor` and `wrapWebSocket`. The reconnector  object is created by `reconnect-core`.
✔️